sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -45,49 +45,62 @@ import org.apache.spark.sql.types._
  * (4, 6)
  *
  * @param analyzer analyzer to use for processing DEFAULT values stored as text.
- * @param catalog  the catalog to use for looking up the schema of INSERT INTO table objects.
  */
-case class ResolveDefaultColumns(
-  analyzer: Analyzer,
-  catalog: SessionCatalog) extends Rule[LogicalPlan] {
+case class ResolveDefaultColumns(analyzer: Analyzer) extends Rule[LogicalPlan] {
 
   // This field stores the enclosing INSERT INTO command, once we find one.
   var enclosingInsert: Option[InsertIntoStatement] = None
   // This field stores the schema of the target table of the above command.
   var insertTableSchemaWithoutPartitionColumns: Option[StructType] = None
+  // This field records if we've replaced an expression, useful for skipping unneeded copies.
```

Comment:
Is this change related to the alter column change? If not, let's open a separate PR for it.

Reply:
This field replaces the usage of `Option[LogicalPlan]` in all the methods below. When adding more logic, I found that the method call sequence got very complicated when checking the optional values. This PR adds more methods to the sequence, so it is related :)

Reply:
Sorry I misunderstood your comment. I will prepare a separate refactoring-only PR for this change and send it out separately.

Reply:
This refactoring-only PR is now ready for review at: https://github.com/apache/spark/pull/36398.

Reply:
Let's drop the changes of this file in this PR

Reply:
SG, I integrated in the recent refactoring changes and pushed the result to this PR. This `ResolveDefaultColumns` file is no longer changed in this PR anymore.